strawberryfields/gbs/plot.py

```diff
@@ -345,49 +346,72 @@ def points(
     return f
 
 
-LIGHT_BLUE = "#94cecf"
-
-BAR_COLOUR = LIGHT_BLUE
-
-
-def spectrum(energies: list, bins: Optional[int] = 0):  # pragma: no cover
-    """Plots a spectrum based on input sampled energies.
+def spectrum(
+    energies: list, gamma: float = 100.0, xmin: float = None, xmax: float = None
+):  # pragma: no cover
+    """Plots a vibronic spectrum based on input sampled energies.
 
     Args:
         energies (list[float]): a list of sampled energies
-        bins (int): maximum number of bins. If zero, the number of bins is decided automatically
-            by Plotly.
+        gamma (float): parameter specifying the width of the Lorentzian function
+        xmin (float): minimum limit of the x axis
+        xmax (float): maximum limit of the x axis
 
     Returns:
-         Figure: spectrum in the form of a histogram of energies
+         Figure: spectrum in the form of a histogram of energies with a Lorentzian-function curve
     """
     try:
         import plotly.graph_objects as go
     except ImportError:
         raise ImportError(plotly_error)
 
-    h = go.Histogram(x=energies, nbinsx=bins, marker={"color": BAR_COLOUR})
+    emin = min(energies)
+    emax = max(energies)
+    if xmin is None:
+        xmin = emin - 0.1 * (emax - emin)
+    if xmax is None:
+        xmax = emax + 0.1 * (emax - emin)
+    bins = int(emax - emin) // 5
```

Comment:
So now we hardcode-in the number of bins such that each bin corresponds to a width of 5 in the x-axis?
Not sure whether now or later, but would be good to allow the user to specify the number of bins

Reply:
Actually this might be a case where adding a comment is necessary as I find this line and the one below not so clear

Reply:
The main problem with making the bin size optional is potential inconsistency in the position of bars and Lorentzian peaks. Note that the Lorentzian peaks appear at the exact position of the energies but the bars are bin dependent. For example, if you have one count at an energy value of 1000, the Lorentzian peak appears at 1000 while using large bins shifts the location of the bars.